src/lexer/tiger.lex

```diff
@@ -1,7 +1,10 @@
 type pos = int
 type lexresult = Token.token
 
-fun eof() = Token.EOF
+fun eof() =
+    (if SrcString.isBuildingString() then raise SrcString.StringNotClosed(Newline.getPos(SrcString.getStartPos()), Newline.getLine(SrcString.getStartPos()))
+    else if not (SrcComment.closed()) then raise SrcComment.CommentNotClosed(Newline.getPos(hd(SrcComment.getComments())), Newline.getLine(hd(SrcComment.getComments())))
```

Comment:
I like the idea of using ```ErrorMsg```. I see us having two options here. First, we can only detect a single error and exit at the lexer(what is already going on) or second, we can parse up to the unclosed string/comment and maintain state that the lexer failed.